editor/store/selectors.js

```diff
@@ -1597,24 +1720,18 @@ export function getBlockListSettings( state, uid ) {
  *
  * @return {string[]|boolean} Blocks that can be nested inside the block with the specified uid, or true/false to enable/disable all types.
  */
+// Disable reason: We have to accept `globallyEnabledBlockTypes` so as to be backwards compatible
```

Comment:
I don't understand. Can you explain what will break if `globallyEnabledBlockTypes` were removed as an argument here?

Reply:
Nothing will break, but it masks the fact that this method accepts `globallyEnabledBlockTypes` as its third argument and must continue to do until it is removed in 3.2.

For example, say that we remove the argument. Now, imagine that a developer wants to add a new `sortBy` argument in version 3.0. They would likely change the signature to look like this:

```jsx
function getSupportedBlocks( state, uid, sortBy = 'asc' )
```

The developer assumes that this is a non-breaking change, since they are just adding a new optional argument. But this is incorrect, since it will break any pre-3.0 code that makes calls such as `wp.data.select( 'core/editor' ).getSupportedBlocks( 'abc123', [ 'core/paragraph' ] )`.

Reply:
Sounds reasonable. My question was mostly for...

- Ensuring we don't keep dead code around (the reason the rule exists in the first place)
- Clarifying wording, since it implies that it is necessary for some reason, though it is not used ("We have to accept...")